### PR TITLE
ENT-618 - add update eca doku

### PR DESCRIPTION
**Pull request by Zylinder**

### CODE REVIEW

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+indexed: false
+---
+
+This guide describes how you update the Enterprise Client Administration.
+
+<div class="toc-list"></div>
+
+## How to update the ECA
+
+### 1. The back up the ECA.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: Backup the ECA

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+First of all, you have to[back up](/enterprise/tech-guide/backup-eca-guide/)the eca.
+
+For this we have already a guide. [Back up the eca](/enterprise/tech-guide/backup-eca-guide/)
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: For this usecase we already provide a userguide.

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+----------------------
+
+### 2. Install the ECA
+
+After you have created a backup of your eca, you can install the new version.
+
+This is very easy.
+
+Just [download](/enterprise/tech-guide/installation-guide/) a new version of the eca and then[install](/enterprise/tech-guide/installation-guide/)it without deleting anything in the same directory.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: the updated version

Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: and then extract all files into your existing directory

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+[Installation Guide](/enterprise/tech-guide/installation-guide/)
+
+After you have done this, you are finished and now you are having a updated version of the eca.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: After you completed this step you are finished.

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+----------------------
+
+### 2. Install the ECA
+
+After you have created a backup of your ECA, you can install the new version.
+
+This is very easy.
+
+Just [download](/enterprise/tech-guide/installation-guide/) the updated version of the ECA and and then extract all files into your existing directory.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: the ECA Update package (sorry my mistake)

extract all files into your existing eca root directory

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+Now, you can install the `setup.phar`.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: Execute the setup.phar and follow our installation guide

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+----------------------
+
+### 2. Install the ECA
+
+After you have created a backup of your ECA, you can install the new version.
+
+This is very easy.
+
+Just [download](/enterprise/tech-guide/installation-guide/) the updated version of the the ECA Update package and and then extract all files into your existing directory.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: and

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+<div class="toc-list"></div>
+
+## How to update the ECA
+
+----------------------
+
+### 1. Start the maintenance mode
+
+First of all, you should turn the eca into the maintenance mode.
+This blocks the frontend for every user interaction, so that no invalid state can be created when the update is happening.
```
Comment (JanPietrzyk) [inline source/enterprise/tech-guide/update-eca-guide.md]: *invalid state* might be a little too much

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+For this just create a app.lock file in the root of your eca directory.
+
+E.g.
+`/var/www/enterprise-client-administration/app.lock`
```
Comment (JanPietrzyk) [inline source/enterprise/tech-guide/update-eca-guide.md]: This does not create a file. (Hint: `touch`)

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+----------------------
+
+### 2. Turn of the supervisor
+
+Now you can turn of the supervisor.
+
+Just start a terminal and execute
+
+```shell
+sudo supervisorctl stop
```
Comment (JanPietrzyk) [inline source/enterprise/tech-guide/update-eca-guide.md]: This fails, please use `sudo supervisorctl stop all`

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+### 5. Turn back the ECA into production
+
+Now you can execute 
+
+```shell
+sudo supervisorctl restart
+```
+
+in the terminal to restart supervisor.
```
Comment (JanPietrzyk) [inline source/enterprise/tech-guide/update-eca-guide.md]: ... to restart the eca background processes

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+For this just create a app.lock file in the root of your eca directory.
+
+E.g.
+`/var/www/enterprise-client-administration/app.lock`
+
+----------------------
+
+### 2. Turn of the supervisor
+
+Now you can turn of the supervisor.
```
Comment (JanPietrzyk) [inline source/enterprise/tech-guide/update-eca-guide.md]: ... the ECA background processes, by stopping the supervisor service

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+
+in the terminal to restart supervisor.
+
+Them you can turn off the maintenance mode, so the frontend is accessible.
+
+```shell
+rm /var/www/enterprise-client-administration/app.lock
+```
+
+After you completed this steps you are finished.
```
Comment (JanPietrzyk) [inline source/enterprise/tech-guide/update-eca-guide.md]: Sth like: "Open Frontend, validate the update, validate that the processes are working" would be nice

File: source/enterprise/tech-guide/update-eca-guide.md (hunk 8 above)
Comment (Zylinder) [inline source/enterprise/tech-guide/update-eca-guide.md]: from

Comment (Zylinder) [inline source/enterprise/tech-guide/update-eca-guide.md]: for secutity reasosns.. frontend blocks

File: source/enterprise/tech-guide/update-eca-guide.md (hunk 9 above)
Comment (Zylinder) [inline source/enterprise/tech-guide/update-eca-guide.md]: this should show the path not the file

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+## How to update the ECA
+
+----------------------
+
+### 1. Start the maintenance mode
+
+First of all, you should turn the eca into the maintenance mode for security reasons.
+This blocks the frontend from every user interaction.
+
+For this just create a app.lock file in the root of your eca directory.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: To enable the maintenance mode create a `app.lock` file in the root directory of your eca.

File: source/enterprise/tech-guide/update-eca-guide.md (hunk 10 above)
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: still missing

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+
+Now you can execute 
+
+```shell
+sudo supervisorctl restart
+```
+
+in the terminal to restart the ECA background processes.
+
+Them you can turn off the maintenance mode, so the frontend is accessible.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: Them

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+
+in the terminal to restart the ECA background processes.
+
+Then you can turn off the maintenance mode, so the frontend is accessible.
+
+```shell
+rm /var/www/enterprise-client-administration/app.lock
+```
+
+After you completed this steps you should open the frontend and validate the update and that the processes are working.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: this

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+## How to update the ECA
+
+----------------------
+
+### 1. Start the maintenance mode
+
+First of all, you should turn the eca into the maintenance mode for security reasons.
+This blocks the frontend from every user interaction.
+
+To enable the maintenance mode create a app.lock file in the root directory of your eca.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: Wrapp app.lock into `

File: source/enterprise/tech-guide/update-eca-guide.md [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+
+in the terminal to restart the ECA background processes.
+
+Then you can turn off the maintenance mode, so the frontend is accessible.
+
+```shell
+rm /var/www/enterprise-client-administration/app.lock
+```
+
+After you completed this steps you should open the frontend and validate this update and that the processes are working.
```
Comment (linushstge) [inline source/enterprise/tech-guide/update-eca-guide.md]: this is singular

**Pull request merged**